mod-di-converter-storage-server/src/main/java/org/folio/services/importprofile/ProfileImportServiceImpl.java

```diff
@@ -0,0 +1,167 @@
+  private final ProfileSnapshotService profileSnapshotService;
+
+  public ProfileImportServiceImpl(@Autowired ProfileSnapshotService profileSnapshotService,
+                                  @Autowired ProfileService<JobProfile, JobProfileCollection, JobProfileUpdateDto> jobProfileService,
+                                  @Autowired ProfileService<MatchProfile, MatchProfileCollection, MatchProfileUpdateDto> matchProfileService,
+                                  @Autowired ProfileService<ActionProfile, ActionProfileCollection, ActionProfileUpdateDto> actionProfileService,
+                                  @Autowired ProfileService<MappingProfile, MappingProfileCollection, MappingProfileUpdateDto> mappingProfileService) {
+    this.profileSnapshotService = profileSnapshotService;
+
+    profileTypeToSaveFunction = new EnumMap<>(ProfileType.class);
+
+    profileTypeToSaveFunction.put(MAPPING_PROFILE, (snapshot, okapiParams) -> mappingProfileService.saveProfile(new MappingProfileUpdateDto()
+      .withProfile((MappingProfile) snapshot.getContent()), okapiParams).map(p -> p));
+
+    profileTypeToSaveFunction.put(ACTION_PROFILE, (snapshot, okapiParams) -> actionProfileService.saveProfile(new ActionProfileUpdateDto()
+      .withProfile((ActionProfile) snapshot.getContent())
+      .withAddedRelations(formAddedRelations(snapshot, ACTION_PROFILE)), okapiParams).map(p -> p));
+
+    profileTypeToSaveFunction.put(MATCH_PROFILE, (snapshot, okapiParams) -> matchProfileService.saveProfile(new MatchProfileUpdateDto()
+      .withProfile((MatchProfile) snapshot.getContent()), okapiParams).map(p -> p));
+
+    profileTypeToSaveFunction.put(JOB_PROFILE, (snapshot, okapiParams) -> jobProfileService.saveProfile(new JobProfileUpdateDto()
+      .withProfile((JobProfile) snapshot.getContent())
+      .withAddedRelations(formAddedRelations(snapshot, JOB_PROFILE)), okapiParams).map(p -> p));
+  }
+
+  @Override
+  public Future<ProfileSnapshotWrapper> importProfile(ProfileSnapshotWrapper profileSnapshot, String tenantId, OkapiConnectionParams okapiParams) {
+    if (profileSnapshot.getContentType() != JOB_PROFILE) {
+      String errorMessage = String.format(PROFILE_SNAPSHOT_INVALID_TYPE, profileSnapshot.getContentType(), JOB_PROFILE);
+      LOGGER.warn("importProfile:: {}", errorMessage);
+      return Future.failedFuture(new BadRequestException(errorMessage));
+    }
+    LOGGER.info("importProfile:: Started import for Job Profile with id {}", profileSnapshot.getProfileId());
+
+    try {
+      convertProfileSnapshotWrapperContent(profileSnapshot);
+    } catch (Exception e) {
+      return Future.failedFuture(new BadRequestException(String.format("Cannot map profile content, error: %s", e.getMessage())));
+    }
+
+    var profileTypeToSnapshots = getProfileTypeToSnapshot(profileSnapshot);
+    var profileTypesInSaveOrder = profileTypeToSnapshots.keySet().stream()
+      .sorted(Comparator.comparingInt(IMPORT_PROFILES_ORDER::indexOf))
+      .toList();
```

Comment:
@RomanChernetskyi Let's create a separate method for it. For example "orderProfilesBeforeSaving"

Reply:
Done, please review